java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -410,6 +411,31 @@ private void visit(Tree tree, Tree terminator) {
     if (tree.parent().is(Tree.Kind.EXPRESSION_STATEMENT)) {
       programState = ProgramState.unstack(programState, programState.stack.size()).a;
     }
+    programState = removeUnreachableSymbolicValues(programState);
+  }
+
+  private ProgramState removeUnreachableSymbolicValues(final ProgramState ps) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "removeUnreachableSymbolicValues" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)